qubesadmin/tools/qvm_start_daemon.py

```diff
@@ -653,16 +702,20 @@ def on_connection_established(self, _subject, _event, **_kwargs):
 
             power_state = vm.get_power_state()
             if power_state == 'Running':
-                asyncio.ensure_future(
-                    self.start_gui(vm, monitor_layout=monitor_layout))
-                asyncio.ensure_future(self.start_audio(vm))
+                if "guivm" in self.enabled_services:
+                    asyncio.ensure_future(
+                        self.start_gui(vm, monitor_layout=monitor_layout)
+                    )
+                if "audiovm" in self.enabled_services:
+                    asyncio.ensure_future(self.start_audio(vm))
                 self.xid_cache[vm.name] = vm.xid, vm.stubdom_xid
             elif power_state == 'Transient':
                 # it is still starting, we'll get 'domain-start'
                 # event when fully started
-                if vm.virt_mode == 'hvm':
+                if vm.virt_mode == 'hvm' and "guivm" in self.enabled_services:
```

Comment:
Put the enabled services check first, to avoid accessing virt_mode property if not needed.